File: SocialAppFlutter/social_app/lib/services/auth/get_user_profile_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:social_app/Models/attended_category.dart';
import 'package:social_app/Models/attended_event.dart';
import 'package:social_app/Models/event_and_rating_average.dart';

import '../../Models/auth/basic_response.dart';
import '../../Models/user.dart';

part 'json_generated_code/get_user_profile_response.g.dart';

@JsonSerializable(explicitToJson: true)
class GetUserProfileResponse extends BasicResponse {
  final User user;
  final bool isFriend;
  final List<AttendedEvent> eventsAttended;
  final List<User> admirers;
  final List<EventAndRatingAverage> myOwnEvents;
  final int honors;
  final int givenHonors;
  final List<AttendedCategory> attendedCategories;
  final bool friendRequestSent;

  GetUserProfileResponse(
      int statusCode,
      String message,
      Type type,
      this.user,
      this.isFriend,
      this.eventsAttended,
      this.admirers,
      this.myOwnEvents,
      this.honors,
      this.givenHonors,
      this.attendedCategories, this.friendRequestSent)
      : super(statusCode, message, type);

  factory GetUserProfileResponse.fromJson(Map<String, dynamic> json) =>
      _$GetUserProfileResponseFromJson(json);

  Map<String, dynamic> toJson() => _$GetUserProfileResponseToJson(this);
}

File: APIs/AuthService/AuthService/DTOs/GetUserProfileResponse.cs
namespace AuthService.DTOs
{
    public class GetUserProfileResponse : BasicResponse
    {
        public UserObject User { get; set; } = null!;
        public bool IsFriend { get; set; }
        public IEnumerable<UserObject> Admirers { get; set; } = null!;
        public IEnumerable<AttendedEvent> EventsAttended { get; set; } = null!;
        public IEnumerable<EventObjectAndRatingAverage> MyOwnEvents { get; set; } = null!;
        public int Honors { get; set; }
        public int GivenHonors { get; set; }
        public IEnumerable<AttendedCategory> AttendedCategories { get; set; } = null!;
        public bool FriendRequestSent { get; set; } = false;
    }
}
